### DISCUSSION TITLE
DDS140 SainSmart

### DISCUSSION THREAD

**Discussion by PM490** [OpenHantek/OpenHantek6022#169] [Q&A] [answered]
Thank you for this great effort. Would like to know if the DDS140 is in the roadmap.
I have it running on a Windows 7 laptop and are looking for alternatives to bring it to ubuntu (specifically ubuntu mate 20.04). Installed OpenHantek6022 3.2, but it does not recognize the scope. The device shows as ID 8312:8312 BUUDAI  USB104.
I see it is listed as _planned_ in the sigrok supported hardware (https://sigrok.org/wiki/Supported_hardware#Oscilloscopes).
When available I can contribute with an installation guide (also can assist translating the existing one to spanish)

**Comment by Ho-Ro** [comment, marked as answer]
Hi Pablo,

Short answer: No, the DDS140 will not be supported by the OH6022.

The main reason is the fact that the DDS140 - in contrast to the DDS120 - has a different HW structure. The DDS120 (like the Hantek 6022 devices) transmits the sampled analog data directly unbuffered and in real time via USB, this limits the sampling rate to 48 MS/s (theoretically, practically it can transmit 24..30 MS/s without much dropouts). The DDS140 pre-processes and buffers the data (possibly using an FPGA) to achieve 100 MS/s. This leads to a different command and acquisition structure. Since I do not own a DDS140, I do not feel a personal need to support this device. OH6022 has been created to give my 6022BE a better software. However, this does not work if the target device is not on my desk, the adaptations for 6022BL and DDS120, which have only minimally different HW, needed several loops. The same applies to operating systems other than (Debian) Linux, I don't work with macOS or Windows, so support for these systems is marginal.

Sorry, Martin

**Reply by PM490** [reply]
Thank you Martin, again great job with OH6022 and thank you for sharing it. Understand the challenge with the DDS140.
